eval/topdown.go

```diff
@@ -962,9 +967,66 @@ func evalTerms(ctx *TopDownContext, iter TopDownIterator) error {
 	default:
 		panic(fmt.Sprintf("illegal argument: %v", t))
 	}
+
+	if indexAvailable(ctx, ts) {
+
+		ref, isRef := ts[1].Value.(opalog.Ref)
+
+		if isRef {
+			ok, err := indexBuildLazy(ctx, ref)
+			if err != nil {
+				return errors.Wrapf(err, "index build failed on %v", ref)
+			}
+			if ok {
```

Comment:
Perhaps indexAvailable and indexBuildLazy could be merged. Let me see if this would make it more readable. Otherwise, I'll leave as is.